### PR TITLE
Fallback to `O_DIRECTORY` open_flags if `O_SEARCH` is undefined

### PR DESCRIPTION
**Pull request by johnsonjh**
Fallback to `O_DIRECTORY` if `O_SEARCH` is undefined on AIX and macOS builds. This allows libsir to build on some older systems. Verified on macOS using Xcode 12.1 and IBM AIX 7.1.  This also allows Intel C on Mac to enable IPO (which requires using an old SDK).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,7 +58,7 @@ At this time, libsir is supported (*that is, it compiles and passes the test sui
 | ------:|:--------- |
 | **Linux**&nbsp;≳2.6.32<br>(glibc&nbsp;2.18, musl&nbsp;≳1.2.3, uClibc‑ng&nbsp;1.0.43, Bionic&nbsp;19) | **GCC**&nbsp;(4.8.4&nbsp;‑&nbsp;13.1.1),&nbsp; **Clang**&nbsp;(3.8&nbsp;‑&nbsp;16.0.6),&nbsp; **Oracle&nbsp;Studio&nbsp;C/C++**&nbsp;(≳12.6),&nbsp; **IBM&nbsp;Advance&nbsp;Toolchain**&nbsp;(14&nbsp;‑&nbsp;16),&nbsp; **IBM&nbsp;XL&nbsp;C/C++**&nbsp;(16.1),&nbsp; **IBM&nbsp;Open&nbsp;XL&nbsp;C/C++**&nbsp;(17.1.1),&nbsp; **NVIDIA&nbsp;HPC&nbsp;SDK&nbsp;C/C++**&nbsp;(23.5),&nbsp; **Arm&nbsp;HPC&nbsp;C/C++**&nbsp;(22.1),&nbsp; **AMD&nbsp;Optimizing&nbsp;C/C++**&nbsp;(4.0.0),&nbsp; **Intel&nbsp;oneAPI&nbsp;DPC++/C++**&nbsp;(2023.1),&nbsp; **Intel&nbsp;C++&nbsp;Compiler&nbsp;Classic**&nbsp;(2021.9),&nbsp; **Android&nbsp;NDK**&nbsp;(r25c),&nbsp; **Chibicc**&nbsp;(2020.12.6) |
 | **AIX**&nbsp;7.2 | **GCC**&nbsp;(8.3&nbsp;‑&nbsp;11.3),&nbsp; **IBM&nbsp;XL&nbsp;C/C++**&nbsp;(16.1),&nbsp; **IBM&nbsp;Open&nbsp;XL&nbsp;C/C++**&nbsp;(17.1) |
-| **macOS**&nbsp;≳10.15 | **Xcode**&nbsp;(12.4&nbsp;‑&nbsp;15),&nbsp; **GCC**&nbsp;(10.4&nbsp;‑&nbsp;13.1),&nbsp; **Clang**&nbsp;(12.0.1&nbsp;‑&nbsp;16.0.6),&nbsp; **Intel&nbsp;C++&nbsp;Compiler&nbsp;Classic**&nbsp;(2021.9) |
```
Comment (aremmell) [inline README.md:61]: I have Xcode 14.3 installed here

File: src/sirfilesystem.c [hunk 2]
```diff
@@ -46,7 +46,11 @@ bool _sir_pathgetstat(const char* restrict path, struct stat* restrict st, sir_r
     if (relative) {
 #if !defined(__WIN__)
 # if defined(__MACOS__) || defined(_AIX)
+#  if !defined(O_SEARCH)
```
Comment (aremmell) [inline src/sirfilesystem.c:49]: O_SEARCH should always be defined if _DARWIN_C_SOURCE is, but I have seen this happen on Traivs machines. I have looked at the headers, and I don't understand it.

Have you confirmed that it still works with O_DIRECTORY?

Comment (aremmell) [reply src/sirfilesystem.c:49]: Actually, it's more nuanced. There is a selector for _DARWIN_C_LEVEL in sys/cdefs.h I believe. It has _ANSI_SOURCE and _POSIX_C_SOURCE involved.

Comment (aremmell) [reply src/sirfilesystem.c:49]: And _DARWIN_C_LEVEL has to be _DARWIN_C_FULL

**Review by aremmell — APPROVED** [review]

**Review by aremmell — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (johnsonjh) [reply README.md:61]: 15 is supposed to be final in a few days.

File: src/sirfilesystem.c (hunk 2 above)
Comment (johnsonjh) [reply src/sirfilesystem.c:49]: Yeah, this is ONLY if O_SEARCH isn't defined, and that's going to happen only on some older Xcode releases / they seem to have added it sometime around Xcode 12.  I verified that O_DIRECTORY does indeed work and pass all tests if O_SEARCH isn't available.

**Review by aremmell — APPROVED** [review]

**Pull request merged**